### PR TITLE
Replace Window id's with HashMap

### PR DESCRIPTION
**Pull request by sundaram123krishnan**

### CODE REVIEW

File: lala_bar/src/music_bar.rs [hunk 1]
```diff
@@ -65,8 +65,7 @@ pub struct LalaMusicBar {
     balance_text: String,
     bar_index: SliderIndex,
     launcher: Option<launcher::Launcher>,
-    launcherid: Option<iced::window::Id>,
-    hiddenid: Option<iced::window::Id>,
+    hiddenid_lock: bool,
```
Comment (Decodetalkers) [inline lala_bar/src/music_bar.rs:68]: we do not need this anymore

File: lala_bar/src/music_bar.rs [hunk 2]
```diff
@@ -76,10 +75,11 @@ pub struct LalaMusicBar {
     sender: Option<Sender<NotifyCommand>>,
     quite_mode: bool,
     datetime: DateTime<Local>,
-    calendar_id: Option<iced::window::Id>,
     date: Date,
+    is_calendar_open: bool,
```
Comment (Decodetalkers) [inline lala_bar/src/music_bar.rs:79]: also this is not needed

**Comment by Decodetalkers** [comment]
We should not use HashMap for these windows, because they are singleton window, should only exist one, and the singleton is already controlled by iced_layershell

**Review by Decodetalkers — CHANGES_REQUESTED** [review]
just update the ui.. do not change other logic

**Comment by sundaram123krishnan** [comment]
Had thought of having a refactor for managing the window id's by using a `Hashmap`
After hearing from u, as there are now less new windows to manage, maybe we could stick with individual fields in struct.

**Pull request closed without merging**